packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -150,6 +150,55 @@ class GestureRecognizerFactoryWithHandlers<T extends GestureRecognizer> extends
 /// ** See code in examples/api/lib/widgets/gesture_detector/gesture_detector.1.dart **
 /// {@end-tool}
 ///
+/// ## Troubleshooting
+///
+/// Why isn't my parent [GestureDetector] onTap method called?
+///
+/// Given a parent [GestureDetector] which defines an onTap callback and a child
+/// [GestureDetector] that also defines an onTap callback. When the inner
+/// [GestureDetector] is tapped, both [GestureDetector]s send a [GestureRecognizer]
+/// into the gesture arena because the pointer coordinates are in both
+/// [GestureDetector]s bounds. The child [GestureDetector] wins because it was
+/// the first detector to enter the arena and it got first dips because it is in front.
+/// For more information on gesture desambiguation see:
+/// [Gesture disambiguation](https://docs.flutter.dev/development/ui/advanced/gestures#gesture-disambiguation)
+///
+/// Setting [GestureDetector.behavior] to [HitTestBehavior.opaque]
+/// or [HitTestBehavior.translucent] has no impact on parent-child relationships
+/// like in this case. To block hits on children use an [IgnorePointer] or [AbsorbPointer].
+///
+/// Some callbacks are called during the gesture arena resolution
+/// (for instance onTapDown). If the parent [GestureDetector] defines such callbacks,
+/// they will be called. Some callbacks are conditionnaly called, for instance
+/// [TapGestureRecognizer.onTapDown] triggers once a short timeout has elapsed.
```

Comment:
This paragraph is still a little unclear. What are we actually trying to document here?

Maybe:

Some callbacks (e.g. onTapDown) can fire before a recognizer wins the arena, and others (e.g. onTapCancel) fire even when it loses the arena. Therefore, the parent detector in the example above may call some of its callbacks even though it looses in the arena.

Reply:
This seems to contradict the statements above. 🙃  Should there be a separate example to illustrate the difference? Cases where the parent callback is triggered and cases where it is not?

Reply:
@goderbauer I added this section after playing with the example, I noticed that `onTapDown` callback is not called after a very fast tap. From https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/gestures/tap.dart#L150-L162, it is expected as there is a duration before accepting a PointerDownEvent (the default duration is 100ms).

It might be too detailed as this example aims to explain only the basis. But a watchful user will notice that sometimes the example prints `Green onTapDown` on the console and sometimes not. So I think it’s better to explain this edge case.

@Piinks I don’t think we can have two examples for this because what gives us these two scenarios is how the tap is done : less than 100ms the parent callbacks (onTapDown and onTapCancel) are not called, more than 100ms they are called.